src/math/bandmat.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2015 Michael Yang. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+#![macro_use]
+
+use libc::c_int as c_int;
+use num::traits::NumCast;
+use std::fmt;
+use std::iter::repeat;
+use std::ops::Index;
+use std::slice;
+use vector::ops::Copy;
+use matrix::BandMatrix;
+use Matrix;
+
+#[derive(Debug, PartialEq)]
+pub struct BandMat<T> {
+    rows: usize,
+    cols: usize,
+    sub_diagonals: c_int,
+    sup_diagonals: c_int,
```

Comment:
it feels inconsistent to mix `usize` and `c_int` in that structure

Reply:
On second thought, we could switch the `rows`, `cols`, `sub_diagonals`, and `sup_diagonals` to all use `usize` since it's architecture dependent. Limiting the sizes of the arrays to 32 bits on 64-bit machines (which are most machines nowadays) seems like an unnecessary thing to do.

Reply:
I am all for it! Do you want to include it in this PR or open a separate one?

Reply:
I'll open a separate PR later. As we discussed in Gitter, let's stick to `u32` for now to get it to compile, and then we can expand this later.